### PR TITLE
Add Elasticsearch support

### PR DESCRIPTION
**Pull request by KillerX**
It's enabled using a feature flag.
Indexing runs for both Algolia and ES.
Elastic tests can be run using `make test.elastic` from the `backend` folder.

Every search result has a `searchProvided` field available (if requested in the query) that identifies the source of the results as "algolia" or "elasticsearch", for easier testing/verification where we are getting the results from.

### CODE REVIEW

**Comment by KillerX** [comment]
I think this can be reviewed. The query is working as expected in terms of filtering, but it's probably missing a lot of fine tuning.

**Comment by KillerX** [comment]
@kkuepper Is there a conflict if I put `bccm-*` indexes into the same elastic cluster? Does BMM have a limit by index or does it search all indexes?

**Comment by kkuepper** [comment]
> @kkuepper Is there a conflict if I put `bccm-*` indexes into the same elastic cluster? Does BMM have a limit by index or does it search all indexes?

I think that's fine. BMM only checks the specific index it also creates.

File: backend/search/query_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		},
+	)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, res)
+	assert.Equal(t, 6, res.HitCount)
+
+	// If you want to see the results in detail in terminal
+	//spew.Dump(res)
+}
```
Comment (KillerX) [inline backend/search/query_test.go:60]: @kkuepper can you confirm that this makes sense to you.

Comment (kkuepper) [reply backend/search/query_test.go:60]: I guess. It's the beginning of a test.

**Pull request merged**